providers/active_directory.go

```diff
@@ -181,30 +181,37 @@ func (s *ADProvider) getUserData(username string, password string) (goth.User, e
 		return thisUser, errors.New("Filter matched multiple users")
 	}
 
-    entry := sr.Entries[0]
+	entry := sr.Entries[0]
 
-    if s.config.LDAPEmailAttribute == "" {
-        s.config.LDAPEmailAttribute = "mail"
-    }
+	if s.config.LDAPEmailAttribute == "" {
+		s.config.LDAPEmailAttribute = "mail"
+	}
 
-    if s.config.LDAPAdminUser != "" {
-        bindErr := s.connection.Bind(entry.DN, password)
-        if bindErr != nil {
-            log.Error(ADProviderLogTag+" Bind failed for user: ", username)
-            log.Error(ADProviderLogTag+" --> Error was: ", bindErr)
-            return thisUser, errors.New("Password not matched")
-        }
-        log.Info(ADProviderLogTag+" User bind successful: ", username)
-    }
+	if s.config.LDAPAdminUser != "" {
+		bindErr := s.connection.Bind(entry.DN, password)
+		if bindErr != nil {
+			log.Error(ADProviderLogTag+" Bind failed for user: ", username)
+			log.Error(ADProviderLogTag+" --> Error was: ", bindErr)
+			return thisUser, errors.New("Password not matched")
+		}
+		log.Info(ADProviderLogTag+" User bind successful: ", username)
+	}
 
 	emailFound := false
-    for _, j := range entry.Attributes {
-        if j.Name == s.config.LDAPEmailAttribute {
-            thisUser.Email = j.Values[0]
-            emailFound = true
-            break
-        }
-    }
+	for _, j := range entry.Attributes {
+		if j.Name == s.config.LDAPEmailAttribute {
+			thisUser.Email = j.Values[0]
+			emailFound = true
+		}
+
+		if j.Name == "givenName" {
```

Comment:
@komalsukhani Lets make this claims configurable the same way as LDAPEmailAttribute